libraries/AP_Scripting/applets/arming-checks.lua

```diff
@@ -0,0 +1,624 @@
+-- This script runs arming checks for validations that are important
+-- to some, but might need to behave differently depending on the vehicle or use case
+-- so we don't want to bake them into the firmware. 
+-- The severity of each check can be set by simply setting the value of parameters.
+-- from warnings, which still allow arming to to errors failsafe which prevent arming 
+-- Set the value of the matching parameter to the MAV_SEVERITY of the error level you want.
+--
+-- New checks can optionally be added, by adding a true/false function (false = fail)
+-- and adding a check to the arming_checks table. 
+--
+-- Thanks to @yuri_rage and Peter Barker for help with the Lua and Autotests
+
+SCRIPT_VERSION = "4.7.0-021"
+SCRIPT_NAME = "Arming Checks"
+SCRIPT_NAME_SHORT = "ArmCk"
+
+REFRESH_DELAY = 500      -- wait this many milliseconds between each update loop (while not armed)
+INITIAL_DELAY = 20000   -- wait 20 seconds for the AP to settle down before starting to show messages
+
+GLOBAL_PARAM_TABLE_BASE = 170
```

Comment:
Let's make sure people can find IDs used by searching for the table key string

Reply:
Added PARAM_TABLE_KEY = comments for each key used.